### PR TITLE
docs: update dataset & readme docs

### PR DESCRIPTION
**Pull request by jackapbutler**
Update installation and dataset creation documentation to reflect discussions from [Discord](https://discord.com/channels/850068776544108564/1074753729955381298/1084181164228628572)

### CODE REVIEW

**Comment by jackapbutler** [comment]
Hey @kjappelbaum would you be able to review this one too when you get a chance 🙏

**Review by maw501 — APPROVED** [review]
Nice.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -37,6 +25,7 @@ With "implementing" we mean the following:
 - Make an issue in this repository that you want to add this dataset (we will label this issue and assign it to you)
 - Make a PR that adds in a new folder in `data`
   - `meta.yaml` describing the dataset in the form that `transform.py` produces. We will use this later to construct the prompts.
+    > If your dataset has multiple natural splits (i.e. train, test, validation) you can create a <split>\_meta.yaml for each.
```
Comment (kjappelbaum) [inline CONTRIBUTING.md:28]: I guess this is one way - but I think we handled it differently now in different datasets by adding a `split_col`

**Review by kjappelbaum — COMMENTED** [review]
I only have a comment about the `yaml`s for different splits. Otherwise, great addition, many thanks!

File: CONTRIBUTING.md (hunk 1 above)
Comment (jackapbutler) [reply CONTRIBUTING.md:28]: I think that PR handles the case where datasets have been included in a larger benchmark but my dataset splits are more related to the fact that the dataset itself has natural splits of train, test, validation based on its' usage alone in other papers? 
> If your dataset is part of a benchmark [(here)](https://github.com/OpenBioML/chemnlp/pull/98/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R150) 

If we are specifying the benchmark split in #98 we might also want to add in which benchmark the dataset is a part of so we can remove it from benchmarking? Although there'll be some duplicate information I can't see us handling both cases with the `split_col`.

### TLDR
* `split_col` in #98 indicates the benchmarking split if your dataset is part of a benchmark (but we also don't know the benchmark?)
* `<split>_meta.yaml` in #104 indicates a split in the dataset and handles the case where the dataset alone was used in other papers and we want to maintain that original train,test split.
